src/rdkafka_mock_cgrp.c

```diff
@@ -706,3 +721,930 @@ void rd_kafka_mock_cgrps_connection_closed(rd_kafka_mock_cluster_t *mcluster,
+ * @return `rd_true` if the subscription was changed, if set and different
+ *         from previous one.
+ *
+ * @locks mcluster->lock MUST be held.
+ */
+static rd_bool_t rd_kafka_mock_cgrp_consumer_member_subscribed_topic_names_set(
+    rd_kafka_mock_cgrp_consumer_member_t *member,
+    rd_kafkap_str_t *SubscribedTopicNames,
+    int32_t SubscribedTopicNamesCnt) {
+        rd_bool_t changed = rd_false;
+        if (!SubscribedTopicNames)
+                return changed;
+
+        int32_t i;
+        rd_list_t *new_subscription = rd_list_new(
+            SubscribedTopicNamesCnt > 0 ? SubscribedTopicNamesCnt : 1, rd_free);
+        for (i = 0; i < SubscribedTopicNamesCnt; i++) {
+                rd_list_add(new_subscription,
+                            RD_KAFKAP_STR_DUP(&SubscribedTopicNames[i]));
+        }
+        rd_list_deduplicate(&new_subscription, rd_strcmp2);
+
+        if (!member->subscribed_topics ||
+            rd_list_cmp(new_subscription, member->subscribed_topics,
+                        rd_list_cmp_str)) {
+                if (member->subscribed_topics)
+                        rd_list_destroy(member->subscribed_topics);
+                member->subscribed_topics =
+                    rd_list_copy(new_subscription, rd_list_string_copy, NULL);
+                changed = rd_true;
+        }
+        rd_list_destroy(new_subscription);
+        return changed;
+}
+
+/**
+ * @brief Adds a member to consumer group \p mcgrp. If member with same
+ *        \p MemberId is already present, only updates the connection and
+ *        sets it as active.
+ *
+ * @param mcgrp Consumer group to add the member to.
+ * @param conn Member connection.
+ * @param MemberId Member id.
+ * @param InstanceId Group instance id (optional).
+ * @param session_timeout_ms Session timeout to use.
+ * @param SubscribedTopicNames Array of subscribed topics.
+ *                             Mandatory if the member is a new one.
+ * @param SubscribedTopicNamesCnt Number of elements in \p SubscribedTopicNames.
+ *
+ * @return New or existing member.
+ *
+ * @locks mcluster->lock MUST be held.
+ */
+rd_kafka_mock_cgrp_consumer_member_t *
+rd_kafka_mock_cgrp_consumer_member_add(rd_kafka_mock_cgrp_consumer_t *mcgrp,
+                                       struct rd_kafka_mock_connection_s *conn,
+                                       const rd_kafkap_str_t *MemberId,
+                                       const rd_kafkap_str_t *InstanceId,
+                                       rd_kafkap_str_t *SubscribedTopicNames,
+                                       int32_t SubscribedTopicNamesCnt) {
+        rd_kafka_mock_cgrp_consumer_member_t *member = NULL;
+        rd_bool_t changed                            = rd_false;
+
+        /* Find member */
+        member = rd_kafka_mock_cgrp_consumer_member_find(mcgrp, MemberId);
+        if (!member) {
+                if (SubscribedTopicNamesCnt < 1)
+                        return NULL;
```

Comment:
See previous comment, here in case it's sending an empty subscription it doesn't add the member to the group and returns `RD_KAFKA_RESP_ERR_UNKNOWN_MEMBER_ID`, it's a case that should never happen in the client. An empty assignment is possible, but not an empty subscription. There's this code:
```C
        /* Validate topics */
        if (topics->cnt == 0 || rd_kafka_topic_partition_list_sum(
                                    topics, _invalid_topic_cb, NULL) > 0)
                return RD_KAFKA_RESP_ERR__INVALID_ARG;
```